tests/test_visitors/test_ast/test_classes/test_buggy_super.py

```diff
@@ -0,0 +1,96 @@
+import pytest
+
+from wemake_python_styleguide.violations.oop import BuggySuperContextViolation
+from wemake_python_styleguide.visitors.ast.classes import BuggySuperCallVisitor
+
+error_dict_comprehension = """
+    {super().make_key(it): make_value(it) for it in items}
+"""
+
+error_set_comprehension = """
+    {super().transmute(it) for it in items}
+"""
+
+error_list_comprehension = """
+    [super().transmute(it) for it in items]
+"""
+
+error_generator_expression = """
+    (super().transmute(it) for it in items)
+"""
+
+error_nested_comprehensions = """
+    [
+        it + 1
+        for it in (super().transmute(i) for i in range(10))
+    ]
+"""
+
+error_multiple_nested_comprehensions = """
+    [
+        it + k + el + 1
+        for it in (super().transmute(i) for i in range(10))
+        for k in {v:0 for v in range(10)}
+        for el in (n ** 2 for n in range(10))
+    ]
+"""
+
+correct_dict_comprehension = """
+    {super(cls, self).make_key(it): make_value(it) for it in items}
+"""
+
+correct_set_comprehension = """
+    {super(cls, self).transmute(it) for it in items}
+"""
+
+correct_list_comprehension = """
+    [super(cls, self).transmute(it) for it in items]
+"""
+
+correct_generator_expression = """
+    (super(cls, self).transmute(it) for it in items)
+"""
+
+
+@pytest.mark.parametrize('code', [
+    error_generator_expression,
+    error_nested_comprehensions,
+    error_multiple_nested_comprehensions,
+    error_set_comprehension,
+    error_list_comprehension,
+    error_dict_comprehension,
```

Comment:
What about this?

```python
class B(A):
   def method(self):
       def inner():
           super().method()  # error here
```

Reply:
This is also mentioned in the same bug report.

Reply:
Ah, okay, must have missed this on the first read. Will implement.